### PR TITLE
initial frontend monorepo setup

### PR DESCRIPTION
**Pull request by nsolante**
Setup for frontend mono, let me know if this is the direction you wanted to go or if the setup is fine

### CODE REVIEW

**Review by hazim-j — COMMENTED** [review]
Looks good! Is this all still boilerplate code from Next.js? Could we add prettier and also rename the workspace to `web` instead of `frontend`?

**Comment by nsolante** [comment]
aight, will change from `frontend` to `web`. Yea still pretty much boilerplate code from next.js still on the other pages

File: apps/web/.gitignore [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
+/.pnp
+.pnp.js
+.idea
+.DS_Store
```
Comment (hazim-j) [inline apps/web/.gitignore:8]: can we remove this.

File: .gitignore [hunk 2]
```diff
@@ -5,6 +5,8 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 lerna-debug.log*
+.DS_Store
+.idea
```
Comment (hazim-j) [inline .gitignore:9]: Can we remove this or move `.DS_Store` under a `#misc` section.

**Review by hazim-j — CHANGES_REQUESTED** [review]
Looks good. Some small changes. I reckon let's add prettier to this and then merge. Can tackle the rest of the login flow in next PR.

You can see https://github.com/stackupfinance/checkout for prettier integration example.

**Comment by hazim-j** [comment]
You'll need to add a `yarn run prettier` command to the `web` package so that the `Code format` step in the monorepo workflow can pick it up.

**Comment by nsolante** [comment]
aight ill do all of those changes

File: apps/web/package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  "name": "@stackupfinance/web",
+  "version": "0.1.0",
+  "private": true,
+  "repository": "git@github.com:stackupfinance/stackup.git",
+  "author": "@stackupfinance",
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint",
```
Comment (hazim-j) [inline apps/web/package.json:11]: can you also add a `lint:fix: "eslint . --fix"` command

File: apps/web/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "next/core-web-vitals"
```
Comment (hazim-j) [inline apps/web/.eslintrc.json:2]: You'll need to extend prettier config here.

```json
{
  "extends": ["next", "next/core-web-vitals", "prettier"]
}
```

Comment (hazim-j) [reply apps/web/.eslintrc.json:2]: For reference if you want to integrate prettier with es-lint: https://github.com/prettier/eslint-config-prettier

**Review by hazim-j — CHANGES_REQUESTED** [review]
You'll need two more files:

`.prettierrc.json`: This can just be a blank json for now.
`.eslintignore`: Look at the checkout repo for what you need.

Will need to also sort out `yarn.lock` merge conflict.

**Comment by hazim-j** [comment]
Will be merged in https://github.com/stackupfinance/stackup/pull/67

**Pull request closed without merging**